apps/api/v2/src/ee/calendars/services/ics-feed.service.ts

```diff
@@ -0,0 +1,116 @@
+import { ICSFeedCalendarApp } from "@/ee/calendars/calendars.interface";
+import { CreateIcsFeedOutputResponseDto } from "@/ee/calendars/input/create-ics.output";
+import { CalendarsService } from "@/ee/calendars/services/calendars.service";
+import { CredentialsRepository } from "@/modules/credentials/credentials.repository";
+import { BadRequestException, UnauthorizedException, Logger } from "@nestjs/common";
+import { Injectable } from "@nestjs/common";
+
+import { SUCCESS_STATUS, ICS_CALENDAR_TYPE, ICS_CALENDAR } from "@calcom/platform-constants";
+import { symmetricEncrypt, IcsFeedCalendarService } from "@calcom/platform-libraries";
+
+@Injectable()
+export class IcsFeedService implements ICSFeedCalendarApp {
+  constructor(
+    private readonly calendarsService: CalendarsService,
+    private readonly credentialRepository: CredentialsRepository
+  ) {}
+
+  private logger = new Logger("IcsFeedService");
+
+  async save(
+    userId: number,
+    userEmail: string,
+    urls: string[],
+    readonly = true
+  ): Promise<CreateIcsFeedOutputResponseDto> {
+    return await this.saveCalendarCredentials(userId, userEmail, urls, readonly);
+  }
+
+  async check(userId: number): Promise<{ status: typeof SUCCESS_STATUS }> {
+    return await this.checkIfCalendarConnected(userId);
```

Comment:
We are calling checkIfCalendarConnected only in check so why not move all of the content of checkIfCalendarConnected into check ?